### PR TITLE
fix(@schematics/update): fix strict ssl always as true bug

### PR DESCRIPTION
**Pull request by tengis**
The last release fixes an auth token issue on #10624 However, it doesn't get the correct **strict-ssl** option, since it uses package **rc** - boolean options comes as either boolean or undefined.

### CODE REVIEW

File: packages/schematics/update/update/npm.ts [hunk 1]
```diff
@@ -61,17 +61,20 @@ function getNpmConfigOption(
...
+) {
+  const sslOptions: { strict?: boolean; ca?: Buffer } = {};
 
-  if (strictSsl === 'false') {
-    sslOptions.strict = false;
-  } else if (strictSsl === 'true') {
+  if (strictSsl === undefined) {
     sslOptions.strict = true;
+  } else {
+    sslOptions.strict = <boolean> strictSsl;
```
Comment (clydin) [inline packages/schematics/update/update/npm.ts]: if the `strictSsl` value is either a boolean or undefined then a direct assignment to `sslOptions.strict` should be the only change need (excepting the parameter type change for the function).  The entire conditional structure can be removed.  The underlying library defaults to `true` if the value is not set or undefined.

Comment (tengis) [reply packages/schematics/update/update/npm.ts]: Done :+1:

File: packages/schematics/update/update/npm.ts [hunk 2]
```diff
@@ -61,17 +61,16 @@ function getNpmConfigOption(
   return value;
 }
 
-function getNpmClientSslOptions(strictSsl?: string, cafile?: string) {
-  const sslOptions: { strict?: boolean, ca?: Buffer } = {};
+function getNpmClientSslOptions(
+  strictSsl?: string | boolean | undefined,
```
Comment (clydin) [inline packages/schematics/update/update/npm.ts]: `strictSsl?: boolean`

File: packages/schematics/update/update/npm.ts [hunk 3]
```diff
@@ -61,17 +61,16 @@ function getNpmConfigOption(
...
+  cafile?: string | boolean | undefined,
```
Comment (clydin) [inline packages/schematics/update/update/npm.ts]: `cafile?: string`

File: packages/schematics/update/update/npm.ts [hunk 4]
```diff
@@ -61,17 +61,16 @@ function getNpmConfigOption(
...
+) {
+  const sslOptions: { strict?: boolean; ca?: Buffer } = {};
 
-  if (strictSsl === 'false') {
-    sslOptions.strict = false;
-  } else if (strictSsl === 'true') {
-    sslOptions.strict = true;
-  }
+  sslOptions.strict = <boolean> strictSsl;
```
Comment (clydin) [inline packages/schematics/update/update/npm.ts]: remove cast please

File: packages/schematics/update/update/npm.ts [hunk 5]
```diff
@@ -61,17 +61,16 @@ function getNpmConfigOption(
...
 
   if (cafile) {
-    sslOptions.ca = readFileSync(cafile);
+    sslOptions.ca = readFileSync(<string> cafile);
```
Comment (clydin) [inline packages/schematics/update/update/npm.ts]: remove cast please

Comment (tengis) [reply packages/schematics/update/update/npm.ts]: Cools! I couldn't believe missed them. Thanks

**Comment by clydin** [comment]
Superseded by #11680

**Pull request closed without merging**